File: sdks/php/src/Types/ScenarioUpContext.php
<?php

namespace Autonoma\Sdk\Types;

/**
 * Context passed to a scenario's up.
 *
 * Seed the uniqueness helpers (Unique::uniqueEmail, Unique::uniqueSlug, ...)
 * from testRunId so values are unique per run yet reproducible between up and
 * a later down.
 */
final class ScenarioUpContext
{
    public function __construct(
        public readonly string $testRunId,
    ) {}
}

File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/types/ScenarioUpContext.java
package ai.autonoma.sdk.types;

/**
 * Context passed to a scenario's {@code up}.
 *
 * @param testRunId unique id for this test run - seed the uniqueness helpers
 *                  ({@code UniqueUtil}) from it so values are unique per run
 *                  yet reproducible between up and down.
 */
public record ScenarioUpContext(String testRunId) {}
